src/domains/mapDomain.ml

```diff
@@ -73,6 +73,7 @@ struct
   (* And one less brainy definition *)
   let for_all2 = M.equal
   let equal = for_all2 Range.equal
+  let compare x y = if equal x y then 0 else M.compare Range.compare x y
```

Comment:
Why `if equal x y then 0 else`? If they're not equal, we do the work twice (`equal`, then `compare`).